classes/Kohana/Jam/Price.php

```diff
@@ -250,19 +252,19 @@ public function is($operator, $value)
 		switch ($operator) 
 		{
 			case Jam_Price::EQUAL_TO:
-				$result = ($this->amount() == $value);
+				$result = (abs($this->amount() - $value) < Jam_Price::EPSILON);
 			break;
 			case Jam_Price::GREATER_THAN:
-				$result = ($this->amount() > $value);
+				$result = (($this->amount() - $value) >= Jam_Price::EPSILON);
 			break;
 			case Jam_Price::LESS_THAN:
-				$result = ($this->amount() < $value);
+				$result = (($value - $this->amount()) >= Jam_Price::EPSILON);
 			break;
 			case Jam_Price::GREATER_THAN_OR_EQUAL_TO:
-				$result = ($this->amount() >= $value);
+				$result = (($value - $this->amount()) < Jam_Price::EPSILON);
 			break;
 			case Jam_Price::LESS_THAN_OR_EQUAL_TO:
-				$result = ($this->amount() <= $value);
+				$result = (($this->amount() - $value) < Jam_Price::EPSILON);
```

Comment:
@hkdobrev Yes there is, notice the argument order is switched, if normalised to be the same the GREATER_OR_EQUAL_TO will be ($this->amount() - $value) > - Jam_Price::EPSILON), which is equivalent to abs($this->amount()-$value) < Jam_Price::EPSILON (the EQUAL_TO check) OR $this->amount - $value >= Jam_Price::EPSILON (THE GREATER_THAN check)